File: nysol_python/mcmd/methods/convTex/db/smp/mcross.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mcross'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
item,date,quantity,price
A,20081201,1,10
A,20081202,2,20
A,20081203,3,30
B,20081201,4,40
B,20081203,5,50
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``item`` 項目を単位に ``date`` 項目を横に展開し、
``quantity`` 項目を出力する。
'''
script['sh_code']='''
mcross k=item f=quantity s=date i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcross(k="item", f="quantity", s="date", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='元の入力データに戻す例'
script['comment']='''
例1の出力結果を元に戻すには、同じく ``mcross`` を以下のよう用いればよい。
'''
script['sh_code']='''
mcross k=item f=2008* s=fld a=date i=rsl1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcross(k="item", f="2008*", s="fld", a="date", i="rsl1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='複数の値を出力'
script['comment']='''
``quantity,price`` の2項目を出力する。
'''
script['sh_code']='''
mcross k=item f=quantity,price s=date i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcross(k="item", f="quantity,price", s="date", i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

script={}
script['title']='並びを逆順する例'
script['comment']='''
横に展開する項目名の並びを逆順にする。
'''
script['sh_code']='''
mcross k=item f=quantity,price s=date%r i=dat1.csv o=rsl4.csv
'''
script['py_code']='''
nm.mcross(k="item", f="quantity,price", s="date%r", i="dat1.csv", o="rsl4.csv").run()
'''
script['odatas']='rsl4.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mcross.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
item,date,quantity,price
A,20081201,1,10
A,20081202,2,20
A,20081203,3,30
B,20081201,4,40
B,20081203,5,50
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Expand the array of \verb|date| horizontally and itemize \verb|quantity|
to the corresponding \verb|item|.
EOF
scp=<<'EOF'
more dat1.csv
mcross k=item f=quantity s=date i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Restore the original input data"
comment=<<'EOF'
Restore the output from Example 1 to the original input data with \verb|mcross|.
EOF
scp=<<'EOF'
more rsl1.csv
mcross k=item f=2008* s=fld a=date i=rsl1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Crosstab with multiple fields"
comment=<<'EOF'
Display crosstab results on two fields \verb|quantity,price|.
EOF
scp=<<'EOF'
mcross k=item f=quantity,price s=date i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)


############## Example 4
title="Reverse data sequence"
comment=<<'EOF'
Restore the sequence of the items that was expanded horizontally.
EOF
scp=<<'EOF'
mcross k=item f=quantity,price s=date%r i=dat1.csv o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)
